codegen/README.md

```diff
@@ -0,0 +1,57 @@
+Codegen
+----
+
+Code generation and code processing tools of Helidon.
+
+We see the following three environments that are used for code processing:
+
+1. Annotation processors
+2. Classpath scanning
```

Comment:
I would argue we have these three:

1. Annotation processing.
2. Classpath scanning + Reflection based processing.
3. Build (e.g., maven-plugin) processing.

Reply:
Maven plugin is expected to use classpath scanning, as it cannot use annotation processing. Also we currently do not use reflection based processing AFAIK.